contracts/GenArt721CoreV3.sol

```diff
@@ -629,16 +625,24 @@ contract GenArt721CoreV3 is ERC721Enumerable, IGenArt721CoreContractV3 {
             uint256 royaltyFeeByID
         )
     {
-        artistAddress = projectIdToArtistAddress[tokenIdToProjectId[_tokenId]];
-        additionalPayee = projectIdToAdditionalPayee[
-            tokenIdToProjectId[_tokenId]
-        ];
+        uint256 projectId = _tokenId / ONE_MILLION;
+        artistAddress = projectIdToArtistAddress[projectId];
+        additionalPayee = projectIdToAdditionalPayee[projectId];
         additionalPayeePercentage = projectIdToAdditionalPayeePercentage[
-            tokenIdToProjectId[_tokenId]
-        ];
-        royaltyFeeByID = projectIdToSecondaryMarketRoyaltyPercentage[
-            tokenIdToProjectId[_tokenId]
+            projectId
         ];
+        royaltyFeeByID = projectIdToSecondaryMarketRoyaltyPercentage[projectId];
+    }
+
+    /**
+     * @notice Gets the project ID for a given `_tokenId`.
+     */
+    function tokenIdToProjectId(uint256 _tokenId)
+        public
+        view
+        returns (uint256 _projectId)
+    {
+        return _tokenId / ONE_MILLION;
     }
```

Comment:
nit: let's codify using inline elsewhere in this contract by making this external. should also be pure since ONE_MILLION is a constant.

Reply:
Yessser.